src/persist/src/operators/input.rs

```diff
@@ -162,4 +165,57 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn file_stream_bench() -> Result<(), Error> {
```

Comment:
So "open loop" benchmark is a term I've seen frank use but I'm unfamiliar with it. Can you explain the concept to me? I didn't look at this code closely because I want to understand what it's supposed to be doing first

Also, do you have some objection to criterion for benchmarks? Looks like it's already in the repo so someone has already done the hard work. We could/should also make it generic wrt the persister so we can have both mem and file for easy comparisons

Reply:
Sure! I dunno where the terms come from (I think electronics) but a open loop benchmark is one where theres no feedback from the output back to the thing generating the inputs in a manner that influences the inputs.

So like as a concrete counterexample, lets say I was testing datastructure insert latency or something like that and I had some code like

```
// Example closed loop benchmark
loop {
    let result = foo.insert( ... )
    // crucially, I'm waiting for the result of the insert here
    assert!(result == true)
 }
 ```
 
 lets say I complete 100 inserts, 99 of them take 1ms, but one of them takes 10 seconds. A naive way to measure this would be to say "the average latency is 1ms, and the 99-percentile latency is also 1ms"
 
 A more realistic way to characterize this would be "99.999% of the time spent on this benchmark was spent handling a single request, and so the likely sustainable latency for this system is something on the order of seconds".
 
 In contrast, the benchmark laid out here just pumps data into the dataflow in a predictable rate, and measures latencies (defined as the lag between the amount of time elapsed since running the benchmark vs the number of completed inputs), and gives us a much clearer look into "what is the actual latencies of this system under load"
 
 more details here https://www.slideshare.net/howarddgreen/how-not-to-measure-latency-london-oct-2013

Reply:
No objections to criterion. I wrote this as a cargo test just to move fast